osf_tests/test_preprint.py

```diff
@@ -0,0 +1,2539 @@
+from nose.tools import *  # flake8: noqa (PEP8 asserts)
+import jwe
+import jwt
+import mock
+import furl
+import time
+import urlparse
+import datetime
+from django.utils import timezone
+import pytest
+import pytz
+import itsdangerous
+
+from framework.exceptions import PermissionsError
+from website import settings, mails
+from website.identifiers.utils import get_doi_and_metadata_for_object
+from website.preprints.tasks import format_preprint, update_preprint_share, on_preprint_updated, update_or_create_preprint_identifiers
+from website.project.views.contributor import find_preprint_provider
+from website.util.share import format_user
+from framework.auth import Auth, cas, signing
+from framework.celery_tasks import handlers
+from framework.exceptions import PermissionsError, HTTPError
+from framework.auth.core import Auth
+from addons.osfstorage.models import OsfStorageFile
+from addons.base import views
+from osf.models import Tag, Preprint, PreprintLog, PreprintContributor, Subject, Session
+from osf.exceptions import PreprintStateError, ValidationError, ValidationValueError
+from osf.utils.permissions import READ, WRITE, ADMIN
+from osf_tests.utils import MockShareResponse
+from .utils import assert_datetime_equal
+
+from osf_tests.factories import (
+    ProjectFactory,
+    AuthUserFactory,
+    UserFactory,
+    PreprintFactory,
+    NodeFactory,
+    TagFactory,
+    SubjectFactory,
+    UnregUserFactory,
+    PreprintProviderFactory
+)
+
+from api_tests import utils as api_test_utils
+from tests.utils import assert_preprint_logs
+from tests.base import OsfTestCase
+
+pytestmark = pytest.mark.django_db
+
+@pytest.fixture()
+def user():
+    return UserFactory()
+
+@pytest.fixture()
+def node(user):
+    return NodeFactory(creator=user)
+
+@pytest.fixture()
+def project(user):
+    return ProjectFactory(creator=user)
+
+@pytest.fixture()
+def preprint(user):
+    return PreprintFactory(creator=user)
+
+@pytest.fixture()
+def auth(user):
+    return Auth(user)
+
+@pytest.fixture()
+def subject():
+    return SubjectFactory()
+
+
+class TestPreprint:
```

Comment:
Tests moved over from tests/test_preprint and then a bunch of new tests adapted from osf_tests/test_node.py.